lib/rubocop/cop/discourse/plugins/discourse_event.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Discourse
+      module Plugins
+        # Using `DiscourseEvent.on` leaves the handler enabled when the plugin is disabled.
+        #
+        # @example
+        #   # bad
+        #   DiscourseEvent.on(:event) { do_something }
+        #
+        #   # good
+        #   on(:event) { do_something }
+        #
+        class DiscourseEvent < Base
```

Comment:
Nitpick: the other rule names usually are an action "CallRequiresPlugin", "CheckPluginIsEnabled", etc. Maybe this should be "UsePluginInstanceOn"?

Reply:
I was checking the rule names that are predefined by Rubocop and I see there is no rule surround this, so maybe it is fine?

Reply:
I don’t have a strong opinion on this, but I think you’re probably right, for consistency this should be named like an action :ok_hand: